pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
I'd agree that having multiple providers, one that passes ".docker/config.json" and one that passes ".dockercfg" would be preferable because it blends the keyrings.  This is what I did here: https://github.com/kubernetes/kubernetes/pull/13894

However, I didn't realize this existed when I wrote that.